### PR TITLE
[Improvement-16507] Remove spring-boot-starter-cache

### LINKED ISSUE #16507: [Improvement] Remove spring-boot-starter-cache dependency
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

This is not used anymore.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by binitshrest**
I changed and removed the some extra dependencies that should be removed after removing spring-boot-starter-cache from pom.xml and application.properties file. That has caused the issues. After removing extra dependencies and the server runs smoothly with out any bugs

close #16507

### CODE REVIEW

**Comment by binitshrest** [comment]
> Thanks for opening this pull request! Please check out our contributing guidelines. (https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)
Hi, Can  you please review if i updated based on the given documentations.

**Review by ruanwenjun — COMMENTED** [review]
We should remove the code and dependency of spring-boot-starter-cache.

**Comment by binitshrest** [comment]
> We should remove the code and dependency of spring-boot-starter-cache.

Hi ruanwenjun,
After carefully observing and understanding  the codebase. I have removed the extra dependency of the spring-boot-starter-cache which includes the related annotations like cacheable and cacheconfig. I removed it and the codebase runs without any issues.  I am open for some more suggestions from your side. As spring-boot-starter-cache dependency has been removed from POM.xml and application.properties files. I went some more depth and removed some related annotations.

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/BaseDataSourceParamDTO.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public void setHostAndPortByAddress(String address) {
         }
 
         this.host = String.join(Constants.COMMA, hosts);
-        this.port = Integer.parseInt(portString);
+        this.port = (Integer) Integer.parseInt(portString);
```
Comment (ruanwenjun) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/BaseDataSourceParamDTO.java]: This change is not needed

File: dolphinscheduler-standalone-server/src/main/resources/application.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#
+  #
```
Comment (ruanwenjun) [inline dolphinscheduler-standalone-server/src/main/resources/application.yaml]: This change is not changed.

**Comment by binitshrest** [comment]
@caishunfeng Can you please guide me to fix 5 failing issues?

**Comment by SbloodyS** [comment]
You should check these two dependency. @binitshrest 
```
log4j-api-2.17.2.jar
log4j-to-slf4j-2.17.2.jar
```

**Comment by binitshrest** [comment]
> You should check these two dependency. @binitshrest
> 
> ```
> log4j-api-2.17.2.jar
> log4j-to-slf4j-2.17.2.jar
> ```



> log4j-to-slf4j-2.17.2.jar

Thank you SbloodyS. I found them.  There are so many of these dependencies. Should i remove them too.

**Comment by binitshrest** [comment]
> > You should check these two dependency. @binitshrest
> > ```
> > log4j-api-2.17.2.jar
> > log4j-to-slf4j-2.17.2.jar
> > ```
> 
> > log4j-to-slf4j-2.17.2.jar
> 
> Thank you @SbloodyS. I found them. There are so many of these dependencies. Should i remove them too.

File: dolphinscheduler-api/pom.xml [hunk 3]
```diff
@@ -121,10 +121,6 @@
                     <groupId>org.springframework.boot</groupId>
                     <artifactId>spring-boot-starter-tomcat</artifactId>
                 </exclusion>
-                <exclusion>
-                    <groupId>org.apache.logging.log4j</groupId>
-                    <artifactId>log4j-to-slf4j</artifactId>
-                </exclusion>
```
Comment (ruanwenjun) [inline dolphinscheduler-api/pom.xml]: Why remove this?

File: dolphinscheduler-bom/pom.xml [hunk 4]
```diff
@@ -927,10 +927,6 @@
                         <groupId>org.apache.logging.log4j</groupId>
                         <artifactId>log4j-core</artifactId>
                     </exclusion>
-                    <exclusion>
-                        <groupId>org.apache.logging.log4j</groupId>
-                        <artifactId>log4j-api</artifactId>
-                    </exclusion>
```
Comment (ruanwenjun) [inline dolphinscheduler-bom/pom.xml]: Why remove this?

**Comment by binitshrest** [comment]
Hi @ruanwenjun,
Sure. I will revert this change asap and will work on this bug. I will appreciate a help from you? My project is failing to build with docker setup and other env setup.

**Comment by binitshrest** [comment]
image
Hi @ruanwenjun , the build is succesfully complete.

**Comment by binitshrest** [comment]
@SbloodyS Because of so many vulnerabilities found in this project i.e due to old dependencies of mavens. The codebase is failing to build.  please allow me to change all the dependencies or any suggestion will appreciated.

File: dolphinscheduler-ui/pnpm-lock.yaml [hunk 5]
```diff
@@ -73,7 +57,7 @@ dependencies:
   echarts: 5.4.2
   js-cookie: 3.0.5
   lodash: 4.17.21
-  monaco-editor: 0.34.1
+  monaco-editor: 0.50.0
```
Comment (ruanwenjun) [inline dolphinscheduler-ui/pnpm-lock.yaml]: Please revert this file, this change is not related with this PR.

Comment (binitshrest) [reply dolphinscheduler-ui/pnpm-lock.yaml]: sure.

File: dolphinscheduler-ui/pnpm-lock.yaml [hunk 6]
```diff
@@ -14,7 +14,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Comment (SbloodyS) [inline dolphinscheduler-ui/pnpm-lock.yaml]: Please do not change this.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: dolphinscheduler-ui/pnpm-lock.yaml (hunk 6 above)
Comment (binitshrest) [reply dolphinscheduler-ui/pnpm-lock.yaml]: Sure. On it.

File: .gitignore [hunk 7]
```diff
@@ -55,3 +55,6 @@ dolphinscheduler-master/logs
 dolphinscheduler-api/logs
 __pycache__
 ds_schema_check_test
+.env.txt
+venv
+myenv
```
Comment (ruanwenjun) [inline .gitignore]: Please remove this

**Review by ruanwenjun — COMMENTED** [review]
CI failed, since we have imported `log4j-api` and `log4j-to-slf4j` I didn't find out how to exclude these.

File: .gitignore (hunk 7 above)
Comment (binitshrest) [reply .gitignore]: Sure. Sir. Thank you. I will also try to find a way to exclude these log4j-api and log4j-to-slf4j. I was looking on the same too.

Comment (ruanwenjun) [reply .gitignore]: We need to find out where these two dependencies are imported, and exclude these, otherwise the ci will failed.

Comment (binitshrest) [reply .gitignore]: Sure. Sir.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-s3/pom.xml [hunk 8]
```diff
@@ -50,6 +50,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-s3/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/pom.xml [hunk 9]
```diff
@@ -43,6 +43,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-obs/pom.xml [hunk 10]
```diff
@@ -43,6 +43,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-obs/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/pom.xml [hunk 11]
```diff
@@ -50,6 +50,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/pom.xml [hunk 12]
```diff
@@ -43,6 +43,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-gcs/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-cos/pom.xml [hunk 13]
```diff
@@ -36,6 +36,12 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-cos/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-abs/pom.xml [hunk 14]
```diff
@@ -43,6 +43,12 @@
             <artifactId>dolphinscheduler-common</artifactId>
             <version>${project.version}</version>
             <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-abs/pom.xml]: Do we need to add exclusions here? since this scope is provided.

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-zookeeper/pom.xml [hunk 15]
```diff
@@ -96,6 +96,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-zookeeper/pom.xml:108]: Do we need to add exclusions here? since this scope is test.

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-jdbc/pom.xml [hunk 16]
```diff
@@ -97,6 +97,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-jdbc/pom.xml:109]: Do we need to add exclusions here? since this scope is test.

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-it/pom.xml [hunk 17]
```diff
@@ -35,6 +35,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-it/pom.xml:47]: Do we need to add exclusions here? since this scope is test.

File: dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/pom.xml [hunk 18]
```diff
@@ -74,6 +74,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/pom.xml:86]: Do we need to add exclusions here? since this scope is test.

File: dolphinscheduler-master/pom.xml [hunk 19]
```diff
@@ -283,6 +268,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-master/pom.xml:276]: Do we need to add exclusions here? since this scope is test.

File: dolphinscheduler-extract/dolphinscheduler-extract-worker/pom.xml [hunk 20]
```diff
@@ -34,10 +34,30 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.logging.log4j</groupId>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (ruanwenjun) [inline dolphinscheduler-extract/dolphinscheduler-extract-worker/pom.xml]: Do we need to add exclusions here? since these has been managed by dependencyManagement under root pom.xml.

**Comment by binitshrest** [comment]
@ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.

**Comment by ruanwenjun** [comment]
> @ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.

@binitshrest Yes, the ci has passed. But there are some extra things need to do.
1. Please remove two license files 
```
`dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-api-2.11.2.txt`
`dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-core-2.11.2.txt`
```
Since you have removed these from LICENSE, and we really don't rely on these, good job, you have solved a new problem.
2. I am not sure if we need to add exclusion in all place, I add comment. Since once we add these, it will be hard to remove in the future, my experience is we only need to exclude these in dependencymanagement. It's need to have a try.

**Comment by binitshrest** [comment]
> > @ruanwenjun sir. Did i solve this issue? Please let me know. I am encouraged that all the checks are passed.
> 
> @binitshrest Yes, the ci has passed. But there are some extra things need to do.
> 
> 1. Please remove two license files
> 
> ```
> `dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-api-2.11.2.txt`
> `dolphinscheduler-dist/release-docs/licenses/LICENSE-log4j-core-2.11.2.txt`
> ```
> 
> Since you have removed these from LICENSE, and we really don't rely on these, good job, you have solved a new problem. 2. I am not sure if we need to add exclusion in all place, I add comment. Since once we add these, it will be hard to remove in the future, my experience is we only need to exclude these in dependencymanagement. It's need to have a try.

Yes, Sir. We don't need to add that exclusion code in all place. I saw in this morning. Once i put that exclusion in root dependency. It will remove from all. I am not sure if i should remove from test scope . I will make also some changes what you have said earlier.

**Comment by binitshrest** [comment]
> Please retry analysis of this Pull-Request directly on SonarCloud

Do i have to worry about this sir? @ruanwenjun

**Comment by ruanwenjun** [comment]
> > Please retry analysis of this Pull-Request directly on SonarCloud
> 
> Do i have to worry about this sir? @ruanwenjun

No, this is not related to your change, and will not block this PR.

**Review by ruanwenjun — DISMISSED** [review]

File: .gitignore [hunk 21]
```diff
@@ -54,4 +54,4 @@ dolphinscheduler-worker/logs
 dolphinscheduler-master/logs
 dolphinscheduler-api/logs
 __pycache__
-ds_schema_check_test
+ds_schema_check_test
```
Comment (SbloodyS) [inline .gitignore]: Please avoid unnessnary chanage.

**Review by SbloodyS — COMMENTED** [review]
Generally LGTM. Just some NIT.

File: .gitignore (hunk 21 above)
Comment (SbloodyS) [reply .gitignore]: This is still unaddressed.

**Comment by binitshrest** [comment]
This issue was closed accidentally. I am reopening this issue.

**Comment by binitshrest** [comment]
> This issue was closed accidentally. I am reopening this issue.

Reopening

**Review by SbloodyS — APPROVED** [review]
+1

**Review by ruanwenjun — APPROVED** [review]
+1
